### PR TITLE
Feat/redux query headers

### PR DESCRIPTION
**Pull request by elquespera**
Implement HTTP headers editing & sending with GraphQL Request + refactor and rename some things.

image

### CODE REVIEW

**Review by sa8b — COMMENTED** [review]
Great!

File: components/Documentation.tsx [hunk 1]
```diff
@@ -1,14 +1,17 @@
 import { useState } from 'react';
 import { Transition } from '@headlessui/react';
 import { DocumentIcon } from '@heroicons/react/24/solid';
+import FlatButton from './FlatButton';
```
Comment (Tomas-Zagorskis) [inline components/Documentation.tsx]: FlatButton isn't used anywhere

File: components/QueryHeaders.tsx [hunk 2]
```diff
@@ -0,0 +1,122 @@
...
+    onUpdate(data.id, key, value);
+  };
+
+  useEffect(() => {
+    setKey(data.key);
+    setValue(data.value);
+  }, [data]);
+
+  return (
+    <li className="group relative grid grid-cols-[1fr,1fr,auto,auto] gap-1">
```
Comment (Tomas-Zagorskis) [inline components/QueryHeaders.tsx]: there is one column for grid too many

File: components/QueryHeaders.tsx [hunk 3]
```diff
@@ -0,0 +1,122 @@
...
+        className="text-inherit min-w-min px-2 py-1 bg-inherit"
+        value={value}
+        placeholder={t('value')}
+        onBlur={handleUpdate}
+        onChange={handleValueChange}
+      />
+      <FlatButton
+        round
+        className="group-focus-within:opacity-100 opacity-0"
+        onClick={() => onDelete(data.id)}
```
Comment (Tomas-Zagorskis) [inline components/QueryHeaders.tsx]: not sure if you want to do this, but icon is invisible but still clickable on that space. Maybe it's better to keep it visible?

**Review by Tomas-Zagorskis — APPROVED** [review]
Great job, maybe a few header templates could be selected with an input drop down menu and 
validation for not to be the same header

File: components/Documentation.tsx (hunk 1 above)
Comment (elquespera) [reply components/Documentation.tsx]: Fixed

File: components/QueryHeaders.tsx (hunk 2 above)
Comment (elquespera) [reply components/QueryHeaders.tsx]: Fixed

File: components/QueryHeaders.tsx (hunk 3 above)
Comment (elquespera) [reply components/QueryHeaders.tsx]: You're right, it's funny the way it is :)

**Review by elquespera — COMMENTED** [review]
Fixed some stuff

**Pull request merged**